pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -183,20 +197,34 @@ func (pc *PodCache) onEvent(_, pod *v1.Pod, ev model.Event) error {
 
 // notifyWorkloadHandlers fire workloadInstance handlers for pod
 func (pc *PodCache) notifyWorkloadHandlers(pod *v1.Pod, ev model.Event) {
-	// if no workload handler registered, skip building WorkloadInstance
+	// if no workload handler registered, skpodIPsip building WorkloadInstance
 	if len(pc.c.handlers.GetWorkloadHandlers()) == 0 {
 		return
 	}
 	// fire instance handles for workload
-	ep := NewEndpointBuilder(pc.c, pod).buildIstioEndpoint(pod.Status.PodIP, 0, "", model.AlwaysDiscoverable, model.Healthy)
-	workloadInstance := &model.WorkloadInstance{
-		Name:      pod.Name,
-		Namespace: pod.Namespace,
-		Kind:      model.PodKind,
-		Endpoint:  ep,
-		PortMap:   getPortMap(pod),
+	if len(pod.Status.PodIPs) > 0 {
+		for _, itemIP := range pod.Status.PodIPs {
+			ep := NewEndpointBuilder(pc.c, pod).buildIstioEndpoint(itemIP.IP, 0, "", model.AlwaysDiscoverable, model.Healthy)
+			workloadInstance := &model.WorkloadInstance{
+				Name:      pod.Name,
```

Comment:
They key for workload instance is namespace+name so this is broken

Reply:
Hi @howardjohn, do you mean that there may be duplicated workload Instance based on podIPs? If yes, we may need to change the key for workload instance, am I right?

Reply:
Hi @howardjohn, I think the key you mentioned should be the podKey, not the workload instacne which should be unique based on Istio endpoint.